packer/resource-ami/init.sh

```diff
@@ -0,0 +1,125 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ubuntu_ami="ami-0584bcdcc2b2a82e9"
+ubuntu_os_version="16.04"
+ubuntu_ssh_username="ubuntu"
+
+centos_ami="ami-9887c6e7"
+centos_os_version="7.4"
+centos_ssh_username="centos"
+
+packer_file="packer-conf.json"
+
+#=== FUNCTION ==================================================================
+# NAME: startPacker
+# DESCRIPTION: Validate config json and start packer.
+#===============================================================================
+function startPacker() {
+	packer validate packer-conf.json
+	packer build packer-conf.json
+}
+
+#=== FUNCTION ==================================================================
+# NAME: checkIfExists
+# DESCRIPTION: Check if the provided file exists.
+# PARAMETER 1: File to check 
+#===============================================================================
+function checkIfExists() {
+	file=$1
+	if [ -f "$file" ]
+	then
+		echo "$file found."
+		return 0
+	else
+		echo "$file not found."
+		return 1
+	fi
+}
+
+#=== FUNCTION ==================================================================
+# NAME: checkResources
+# DESCRIPTION: Check if the resources given in resource.txt exists.
+#===============================================================================
+function checkResources() {
+	notFound=0	
+	while read F  ; do
+		checkIfExists resources/$F
+		boolExist=$?
+		if [ "$boolExist" == 1 ]
+		then
+			notFound=1
+		fi		
+	done <resources/resources.txt
+	if [ "$notFound" = "1" ]
+	then
+		echo "Resources are missing.. Can not continue."
+		exit;
+	else 
+		creator_ip=$(ip route get 1.1.1.1 | awk '{print $NF; exit}')
+		export PACKER_AMI_CREATOR_IP=$creator_ip
+		startPacker;
+	fi
+}	
+
+#=== FUNCTION ==================================================================
+# NAME: downloadResourcesFromS3
+# DESCRIPTION: Download relevant artifacts from TestGrid S3 bucket. 
+# The function will download Unix resources and OS-specific resources.
+#===============================================================================
+function downloadResourcesFromS3() {
+	echo "Downloading $os AMI resources from S3 (testgrid-resources/packer directory)"
+	aws s3 sync s3://testgrid-resources/packer/Common resources/
+	aws s3 sync s3://testgrid-resources/packer/Unix resources/
+	aws s3 sync s3://testgrid-resources/packer/$os resources/
+}
+
+#Check if Packer, AWS CLI exists
+command -v packer >/dev/null 2>&1 || { echo >&2 "Packer was not found. Please install Packer first."; exit 1; }
+command -v aws >/dev/null 2>&1 || { echo >&2 "AWS CLI was not found. Please install AWS CLI first."; exit 1; }
+
+echo "Select the OS of the AMI:"
+echo "        1 - Ubuntu $ubuntu_os_version"
+echo "        2 - CentOS $centos_os_version"
+echo "        3 - Windows 2016"
+read os;
+case "$os" in
+    1)
+	os="Ubuntu"
+	export PACKER_SOURCE_AMI_ID=$ubuntu_ami
```

Comment:
One particular advantage of Packer is how easy it is to create AMIs with given content across multiple regions at once. We are not leveraging that fact here. Shall we do a bit RnD in there?

Reply:
+1 Will do. :)